File: liveroom-libs/comp_avparam/src/main/java/com/qlive/avparam/QVideoFrameListener.java
package com.qlive.avparam;

public interface QVideoFrameListener {
    default void onYUVFrameAvailable(
            byte[] data,
            QVideoFrameType type,
            int width,
            int height,
            int rotation,
            long timestampNs
    ) {
    }
    default int onTextureFrameAvailable(
            int textureID,
            QVideoFrameType type,
            int width,
            int height,
            int rotation,
            long timestampNs,
            float[] transformMatrix
    ) {
        return textureID;
    }
}

File: liveroom-libs/comp_avparam/src/main/java/com/qlive/avparam/QVideoFrameListener.kt
package com.qlive.avparam

interface QVideoFrameListener {
    fun onYUVFrameAvailable(
        data: ByteArray,
        type: QVideoFrameType,
        width: Int,
        height: Int,
        rotation: Int,
        timestampNs: Long
    ){}

    fun onTextureFrameAvailable(
        textureID: Int,
        type: QVideoFrameType,
        width: Int,
        height: Int,
        rotation: Int,
        timestampNs: Long,
        transformMatrix: FloatArray?
    ): Int{
        return textureID
    }
}

enum class QVideoFrameType {
    YUV_NV21, TEXTURE_RGB, TEXTURE_OES
}

